http/codegen/service_data.go

```diff
@@ -1467,7 +1518,7 @@ func buildBodyType(svc *service.Data, s *httpdesign.ServiceExpr, e *httpdesign.E
 			}
 			desc = fmt.Sprintf("%s is the type of the %q service %q endpoint HTTP %s body.",
 				varname, svc.Name, e.Name(), ctx)
-			if !marshaled {
+			if svr && req || (!svr && !req && md.ViewedResult == nil) {
```

Comment:
What happens to validations other than required? are they still being checked somewhere?

Reply:
The validations generated in the views package will validate everything along with required.